settings/auth.go

```diff
@@ -0,0 +1,54 @@
+package settings
+
+import (
+	"os"
+)
+
+const (
+	EnvAuthRequired         = "AUTH_REQUIRED"
+	EnvKeycloakHost         = "KEYCLOAK_HOST"
+	EnvKeycloakRealm        = "KEYCLOAK_REALM"
+	EnvKeycloakClientID     = "KEYCLOAK_CLIENT_ID"
+	EnvKeycloakClientSecret = "KEYCLOAK_CLIENT_SECRET"
+	EnvAddonAccessSecret    = "ADDON_ACCESS_SECRET"
+)
+
+type Auth struct {
+	// Auth required
+	Required bool
+	// Keycloak client config
+	Keycloak struct {
+		Host         string
+		Realm        string
+		ClientID     string
+		ClientSecret string
+	}
+	// Addon API access secret
+	AddonAccessSecret string
+}
+
+func (r *Auth) Load() (err error) {
+	var found bool
+	r.Required = getEnvBool(EnvAuthRequired, true)
+	r.Keycloak.Host, found = os.LookupEnv(EnvKeycloakHost)
+	if r.Required && !found {
+		panic("KEYCLOAK_HOST required")
```

Comment:
On one hand, I like the panic.
But, we don't do this anywhere else but perhaps there are no sane defaults.  Thinking about running the hub locally.

Reply:
Sure, I can put together a sample realm with some defaults.